internal/common/metrics.go

```diff
@@ -5,41 +5,25 @@ package common
 
 import (
 	"fmt"
-	"sync"
 	"time"
 
 	"github.com/cactus/go-statsd-client/v5/statsd"
-	"go.uber.org/multierr"
 )
 
 // StartLogWrite starts go routines to write logs to each of the logs that are monitored by CW Agent according to
 // the config provided
-func StartSendingMetrics(receiver string, agentRunDuration time.Duration, dataRate int) error {
```

Comment:
I am fine removing my changes to `StartSendingMetrics()` it is not used by my test changes.
However I thought it was overly complicated.
It looks to me like it is running a goroutine, but then it uses a wait group so it is waiting for the goroutine to complete anyways... therefore it didn't need to run in a goroutine, right?

Reply:
It won't be simple though. The StartSendingMetrics is used to generate loads in X `amount of minutes` and also ensure the waiting of X `amount of minutes` for [CloudWatchAgent to send all the metrics in that time](https://github.com/aws/amazon-cloudwatch-agent-test/blob/main/validator/validators/validator.go#L43).

Reply:
What is the difference between:
```
var wg sync.WaitGroup
func foo() {
    wg.Add(1)
    go func() {
        // do stuff that takes some time
        time.Sleep(time.Minute)
    }
    wg.Wait()
}
```
and
```
func foo() {
    // do stuff that takes some time
    time.Sleep(time.Minute)
}
```

Reply:
You can argue to use `sleep` for X `amount of minutes` but sending `50,000` at the end of Xinutes does not mean it will stop there. (e.g extra seconds when sending it)

Reply:
No, I am not suggesting we add time.Sleep(). It was just an example.
I was just wondering if my code change to this function will cause a change in behavior.
It didn't look like it to me for the reason I tried to explain above.

If you run some code in an anonymous goroutine, but then wait for that goroutine to complete in the same function that launched it... Then you did not really need to run it in a goroutine. Right?